### PR TITLE
[BugFix] array function doest not process const inputs

### LINKED ISSUE #24587: array functions with const inputs causes crash.
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. SELECT  '....'
```
select array_contains_all( cast ('[40360,40361]' as array<int>), 40360);
```
### Real behavior (Required)
crash
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by fzhedu**
Fixes https://github.com/StarRocks/starrocks/issues/24587
may array functions cause BE crassh when inputing const columns. usually happen after https://github.com/StarRocks/starrocks/pull/19793

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Seaven — APPROVED** [review]

File: be/src/exprs/array_functions.cpp [hunk 1]
```diff
@@ -94,14 +94,12 @@ static StatusOr<ColumnPtr> do_array_append(const Column& elements, const UInt32C
...
-    const Column* arg0 = columns[0].get();
-    const Column* arg1 = columns[1].get();
-    if (arg0->only_null()) {
+    if (columns[0]->only_null()) {
         return columns[0];
     }
 
-    arg0 = arg0->has_null() || arg0->is_constant() ? arg0 : ColumnHelper::get_data_column(arg0);
-    arg1 = arg1->has_null() || arg1->is_constant() ? arg1 : ColumnHelper::get_data_column(arg1);
+    const Column* arg0 = ColumnHelper::unpack_and_duplicate_const_column(columns[0]->size(), columns[0]).get();
```
Comment (satanson) [inline be/src/exprs/array_functions.cpp:101]: suggest to implement another branch for constant columns, reference to substr

Comment (fzhedu) [reply be/src/exprs/array_functions.cpp:101]: there are many such cases, we can optimize it later. This PR just avoid crash.

**Review by silverbullet233 — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**